packages/cli/src/lib/commands/generate-command.ts

```diff
@@ -207,9 +218,11 @@ export class GenerateCommand extends BaseCommand {
     }
 
     // Handle section-specific options
+    // Use the first source if it's an array for determining section options
+    const source = Array.isArray(options.source) ? options.source[0] : options.source;
     const sectionSupportedOptions = this.generateDocumentationUseCase.getSectionSupportedOptions({
       cicdPlatform: options.cicd,
-      source: options.source,
+      source,
```

Comment:
Keep the original behavior, pass the source option (string|string[]), the use case should manage that